### PR TITLE
net-voip/captagent: New Package.  Version 6.3.1.

### PR DESCRIPTION
**Pull request by jkroonza**
autoconf patch from upstream (in response to issue report).
gcc10 patches pending on PR.

Package-Manager: Portage-3.0.18, Repoman-3.0.2
Signed-off-by: Jaco Kroon <jaco@uls.co.za>

### CODE REVIEW

**Comment by jkroonza** [comment]
Note:  the code base itself requires some work.  For example, --disable-redis has been fixed to not bother linking against redis, but the database_redis.s file still gets built and installed, but will mostly become no-op in nature.  I'm not going to fix this here but will attempt to push PRs upstream towards the project.

File: net-voip/captagent/captagent-6.3.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools
+
+DESCRIPTION="CaptAgent:  100% Open-Source Packet Capture Agent for HEP"
```
Comment (juippis) [inline net-voip/captagent/captagent-6.3.1.ebuild:8]:
```suggestion
DESCRIPTION="CaptAgent: 100% Open-Source Packet Capture Agent for HEP"
```

File: net-voip/captagent/captagent-6.3.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+HOMEPAGE="https://sipcapture.org/"
```
Comment (juippis) [inline net-voip/captagent/captagent-6.3.1.ebuild:9]:
```suggestion
HOMEPAGE="https://sipcapture.org/ https://github.com/sipcapture/captagent"
```

File: net-voip/captagent/captagent-6.3.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+SRC_URI="https://github.com/sipcapture/captagent/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
```
Comment (juippis) [inline net-voip/captagent/captagent-6.3.1.ebuild:12]: Read the source files under upstreams src/ dir, GPL-3 doesn't seem exactly correct ;)

File: net-voip/captagent/captagent-6.3.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="ipv6 mysql pcre redis ssl uv"
```
Comment (juippis) [inline net-voip/captagent/captagent-6.3.1.ebuild:15]: I can't see `uv` having a function in this ebuild.

File: net-voip/captagent/captagent-6.3.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+PATCHES=(
+	# https://github.com/sipcapture/captagent/pull/239 (should be accepted).
+	"${FILESDIR}/${P}-gcc10.patch"
+	# Already upstreamed for next version.
+	"${FILESDIR}/${P}-configure.patch"
+)
+
+DEPEND="dev-libs/json-c
+	net-libs/libpcap
+
```
Comment (juippis) [inline net-voip/captagent/captagent-6.3.1.ebuild:26]: Could we drop this empty line from here.

File: net-voip/captagent/captagent-6.3.1.ebuild (hunk 2 above)
Comment (jkroonza) [reply net-voip/captagent/captagent-6.3.1.ebuild:9]: Updated locally so long.

File: net-voip/captagent/captagent-6.3.1.ebuild (hunk 1 above)
Comment (jkroonza) [reply net-voip/captagent/captagent-6.3.1.ebuild:8]: Ha!  Old typing rule habits die hard.  Fixed locally so long.

File: net-voip/captagent/captagent-6.3.1.ebuild (hunk 4 above)
Comment (jkroonza) [reply net-voip/captagent/captagent-6.3.1.ebuild:15]: In DEPEND ... but that should be associated with a ./configure option which there is not ... rechecking ./configure, it's a hard dep, moving that accordingly and dropping the IUSE.

File: net-voip/captagent/captagent-6.3.1.ebuild (hunk 5 above)
Comment (jkroonza) [reply net-voip/captagent/captagent-6.3.1.ebuild:26]: Done.  I just prefer having a blank line between the unconditional and conditional depends, but that's personal preference, if the policy is to not have it, so be it.

File: net-voip/captagent/captagent-6.3.1.ebuild (hunk 3 above)
Comment (jkroonza) [reply net-voip/captagent/captagent-6.3.1.ebuild]: Updated to GPL-3+

The Bison generated file doesn't specify a version, which means any version applies (as I understand).

There is also one file supplied by Aladdin, but it basically says for any purpose 'as-is' so it's not the standard a Aladdin license and I'm having trouble finding an associated one in ::gentoo/licenses/

Comment (jkroonza) [reply net-voip/captagent/captagent-6.3.1.ebuild]: Updated to GPL-3+ which is more in line with most of the source files.  Two notable exceptions:

capplan.tab.c (generated from .y) which doesn't mention a version.
md5.c which basically says public domain as I interpret it.

I'm happy that we're allowed to redistribute it as part of a GPL-3+ project.  Please confirm.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**